### PR TITLE
Merge wiki content into doc/ directory

### PR DESCRIPTION
**Pull request by rvagg**
Ref: #2435

The proposal here is to deprecate the wiki and replace the current docs with tombstones that point to their copies in the docs/ directory. Then documentation can be managed, organised and edited properly through the GitHub process, like we've done (quite effectively) with the macOS_Catalina.md (h/t to @cclauss for the great work on that).

GitHub wiki's are garbage for broad collaboration and mostly out-of-sight for regular contributors. We have the problem of having a wiki with some docs that are regularly viewed by users thanks to search results but not much effort goes into maintaining them.

This PR pulls all of the wiki content into docs/ including the entire commit history of the wiki, amending each commit with `doc(wiki): `. No other editing has taken place, the proposal is that the first step is to just import & tombstone, then editing and curation can take place after that (editing, removal, renaming, adding, etc.).

### CODE REVIEW

File: docs/Common-issues.md
Comment (cclauss) [inline docs/Common-issues.md:3]:
```suggestion
Make sure you are using the native Python version in OSX.  If you use a MacPorts or HomeBrew version, you may run into problems.
```
https://www.macports.org and https://brew.sh are two separate macOS package managers.

I think we should remove these two paragraphs because Apple what’s stated that native Python will be removed from macOS.  I have been using brew Python from the beginning with zero issues and have seen no one complaining about `execvp`.

File: docs/Common-issues.md
Comment (cclauss) [inline docs/Common-issues.md:7]: This is about node-gyp v0.91 yet the current node-gyp is v8.1.  Is there any evidence that this problem still exists?

**Comment by owl-from-hogvarts** [comment]
Also would be cool to tell in readme where to find docs

**Comment by rvagg** [comment]
I've added a brief note in the README pointing to this directory.

I didn't really want to get into the weeds editing the content, but perhaps some of it just needs to be retired completely. Perhaps the best way to do this is to just edit my branch directly, or PR against my branch for any changes that you think are necessary?

Home.md could probably be renamed to README.md. We could probably improve the filenames (the `"` in the binding.gyp file and the casualness of the names are probably not ideal). I'm not sure the "undeclared identifier" doc is necessary. Common-issues is ... maybe not so common. Maybe we even move the Catalina doc into this directory? (But do we leave a tombstone with a link to the new location for a while since it's probably got some link-fu out in the web).

tbh I don't really have time to make a big job of this but would appreciate help from others to make this land-able and retire the wiki if others see that as a worthy goal.

File: docs/Common-issues.md
Comment (cclauss) [reply docs/Common-issues.md:3]: Let’s agree to land this first and then resolve these issues in future PR.

**Review by cclauss — APPROVED** [review]
Nice!  Thanks for doing this.

**Review by richardlau — APPROVED** [review]
SGTM. No objections to the plan to migrate off the wiki.

I think these docs will end up being included in node-gyp releases, which I don't personally have an issue with but feel like I should point out in case anyone else does.

**Comment by owl-from-hogvarts** [comment]
I think we can somwhow force releas please to ingore them

**Comment by rvagg** [comment]
Well, if you want to help get them up to scratch before we cut another release then it should be ok to merge now work on them immediately afterward. My preference would be to include them in releases, just not necessarily in their current form since they're not all helpful.

**Comment by cclauss** [comment]
> I think these docs will end up being included in node-gyp releases

`.gitignore` to the rescue?

**Comment by richardlau** [comment]
> > I think these docs will end up being included in node-gyp releases
> 
> `.gitignore` to the rescue?

I imagine if we're adding these docs to the repo so they can be edited via git commits/GitHub pull requests we do not want them git ignored :slightly_smiling_face:.

**Comment by rvagg** [comment]
doh! the CI failure is real:

```
  "C:\Program Files\Git\bin\git.exe" checkout --progress --force -B rvagg/wiki-to-doc refs/remotes/origin/rvagg/wiki-to-doc
  Error: error: invalid path 'docs/"binding.gyp"-files-out-in-the-wild.md'
  Error: The process 'C:\Program Files\Git\bin\git.exe' failed with exit code 128
```

will have to change that filename before merging

**Comment by rvagg** [comment]
renamed 3 docs to not have quotes, removed the TypedArray doc which is unhelpful

**Comment by owl-from-hogvarts** [comment]
i think https://github.com/nodejs/node-gyp/issues/2272 mast be added to docs

**Comment by cclauss** [comment]
The idea was to land this PR without further mods and then rebase PRs like #2272 afterwards.

**Review by gengjiawen — APPROVED** [review]

**Comment by owl-from-hogvarts** [comment]
what is needed to land this PR?

**Comment by rvagg** [comment]
@owl-from-hogvarts just waiting after my last push to see if there were any objections. I don't see any so let's make this happen and get on with it!

**Comment by rvagg** [comment]
Merged and all Wiki pages updated to point to that directory. The wiki still remains editable by people with commit access here if anyone thinks there's a better way to communicate the move. I didn't point to specific documents, just the directory, so we have the freedom in the short term to rename and move things around.

Let's hold off on cutting a new release and shipping half baked docs until we're slightly more comfortable with them. But **please contribute your excellent documentation to help users of this project and so we have better docs to point people to when they show up asking questions on a daily basis**.

**Pull request merged**
